lib/absinthe/adapter.ex

```diff
@@ -96,82 +98,131 @@ defmodule Absinthe.Adapter do
       end
-      defp load_ast_node(%Language.SelectionSet{} = node) do
+      def adapt(%Language.SelectionSet{} = node, adaptation) do
         %{node |
-          selections: node.selections |> Enum.map(&load_ast_node/1)}
+          selections: node.selections |> Enum.map(&adapt(&1, adaptation))}
       end
-      defp load_ast_node(%Language.Field{} = node) do
+      def adapt(%Language.Field{} = node, adaptation) do
         %{node |
-          name: node.name |> to_internal_name(:field),
-          selection_set: load_ast_node(node.selection_set),
-          arguments: node.arguments |> Enum.map(&load_ast_node/1),
-          directives: node.directives |> Enum.map(&load_ast_node/1),
+          name: node.name |> do_adapt(:field, adaptation),
+          selection_set: adapt(node.selection_set, adaptation),
+          arguments: node.arguments |> Enum.map(&adapt(&1, adaptation)),
+          directives: node.directives |> Enum.map(&adapt(&1, adaptation)),
         }
       end
-      defp load_ast_node(%Language.ObjectValue{} = node) do
+      def adapt(%Language.FieldDefinition{} = node, adaptation) do
         %{node |
-          fields: node.fields |> Enum.map(&load_ast_node/1)}
+          name: node.name |> do_adapt(:field, adaptation),
+          arguments: node.arguments |> Enum.map(&adapt(&1, adaptation)),
+         }
       end
-      defp load_ast_node(%Language.ObjectField{} = node) do
+      def adapt(%Language.InputValueDefinition{} = node, adaptation) do
         %{node |
-          name: node.name |> to_internal_name(:field),
-          value: load_ast_node(node.value)}
+          name: node.name |> do_adapt(:field, adaptation),
+         }
       end
-      defp load_ast_node(%Language.Argument{} = node) do
+      def adapt(%Language.ObjectValue{} = node, adaptation) do
         %{node |
-          name: node.name |> to_internal_name(:argument),
-          value: load_ast_node(node.value)}
+          fields: node.fields |> Enum.map(&adapt(&1, adaptation))}
       end
-      defp load_ast_node(%Language.ListValue{} = node) do
-        %{node | values: Enum.map(node.values, &load_ast_node/1)}
+      def adapt(%Language.ObjectField{} = node, adaptation) do
+        %{node |
+          name: node.name |> do_adapt(:field, adaptation),
+          value: adapt(node.value, adaptation)}
       end
-      defp load_ast_node(other) do
+      def adapt(%Language.Argument{} = node, adaptation) do
+        %{node |
+          name: node.name |> do_adapt(:argument, adaptation),
+          value: adapt(node.value, adaptation)}
+      end
+      def adapt(%Language.ListValue{} = node, adaptation) do
+        %{node | values: Enum.map(node.values, &adapt(&1, adaptation))}
+      end
+      def adapt(%{__struct__: str} = node, _) when str in @ignore_nodes do
+        node
+      end
+      def adapt(nil, _) do
+        nil
+      end
+      def adapt(other, _) do
+        Logger.warn "Could not adapt #{inspect other}"
```

Comment:
Yeah I think it makes sense here. I do think this should be `"Absinthe: Could not adapt"` because it may be confusing if it shows up in the wild for folks.